TCAT/Controllers/FavoritesViewController.swift

```diff
@@ -0,0 +1,149 @@
+        favoritesCollectionView.register(FavoritesCollectionViewCell.self, forCellWithReuseIdentifier: favoritesReuseIdentifier)
+        favoritesCollectionView.backgroundColor = .clear
+        view.addSubview(favoritesCollectionView)
+    }
+
+    func setupTab() {
+        tab.backgroundColor = Colors.metadataIcon
+        tab.layer.cornerRadius = tabSize.height / 2
+        tab.clipsToBounds = true
+        view.addSubview(tab)
+    }
+
+    private func setupConstraints() {
+
+        let tabTopInset: CGFloat = 6
+        let horizontalPadding = 16
+        let titleBarTopPadding = 21
+        let favoritesTopPadding = 24
+        let bottomPadding = 18
+
+
+        favoritesTitleLabel.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(horizontalPadding)
+            make.top.equalToSuperview().inset(titleBarTopPadding)
+        }
+
+        editButton.snp.makeConstraints { make in
+            make.trailing.equalToSuperview().inset(horizontalPadding)
+            make.top.equalToSuperview().inset(titleBarTopPadding)
+        }
+
+        favoritesCollectionView.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(horizontalPadding)
+            make.trailing.equalToSuperview().inset(horizontalPadding).priority(.high)
+            make.top.equalTo(favoritesTitleLabel.snp.bottom).offset(favoritesTopPadding)
+            make.bottom.equalToSuperview().offset(-bottomPadding).priority(.high)
+        }
+
+        tab.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalToSuperview().inset(tabTopInset)
+            make.size.equalTo(tabSize)
+        }
+
+
+    }
+
+}
+
+extension FavoritesViewController: UICollectionViewDataSource, UICollectionViewDelegate {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == favoritesCollectionView {
+            return favorites.count
+        }
+        return 0
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == favoritesCollectionView {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: favoritesReuseIdentifier, for: indexPath) as! FavoritesCollectionViewCell
+            cell.configure(for: favorites[indexPath.row])
+            return cell
+        }
+        return UICollectionViewCell()
+    }
+
+}
+
+extension FavoritesViewController: UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: 80, height: 95)
+    }
+}
+
+extension FavoritesViewController: PulleyDrawerViewControllerDelegate {
+
+    func partialRevealDrawerHeight(bottomSafeArea: CGFloat) -> CGFloat {
+        return 200
```

Comment:
Is this going to be dynamic?

Reply:
I calculated 200 just from Femi's designs! Do we want this to be dynamic based on the height of a user's phone?

Reply:
I would think we should maintain the aspect ratio of the version in Femi's iPhone Xs designs depending on the device width.

Reply:
Maybe a stupid question but if I make the height of the view controller dynamic, then I'd also have to update my vertical constraints on my views no? When I tried using the aspect ratio from the design, it doesn't always look very good on other devices?